pkg/ranger/envelope.go

```diff
@@ -0,0 +1,105 @@
+package ranger
```

Comment:
nit: Would be nice to have a brief package comment stating what this package does (it isn't obvious).